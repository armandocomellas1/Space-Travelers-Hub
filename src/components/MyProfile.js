import { useSelector, useDispatch } from 'react-redux';
import { useEffect } from 'react';
import './profileStyle.css';
import axios from 'axios';
import ProfileRockets from './MyProfrender';

const MyProfile = () => {
  const dispatch = useDispatch();
  const loading = useSelector((store) => store.rockets.status);
  const rockets = useSelector((state) => state.rockets);
  // console.log('myProfile rockets', rockets);
  // useEffect(() => {
  //   if (loading === loadingStatus.idle) {
  //     dispatch(fecthRockets());
  //   }
  // }, [dispatch, loading]);

  if (rockets.status === 'SUCCEEDED') {
    const rocketsAll = rockets.rockets;
    return (
      <div>
        <h2 className="profile_rockets">My Rockets</h2>
        <div className="profile_main">
          {
            rocketsAll.map((rocket) => {
              if (rocket.reserve === true) {
                return (
                  <ProfileRockets
                    key={`books-lisk-card-${rocket.id}`}
                    {...rocket}
                  />
                );
              }
              return false;
            })
          }
        </div>
      </div>
    );
  }
  return false;
};

export default MyProfile;
