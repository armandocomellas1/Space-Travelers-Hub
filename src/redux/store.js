import { combineReducers } from 'redux';
import { configureStore } from '@reduxjs/toolkit';
import rocketsSlice from './rockets/rocketsApi';
import missionsSlice from './missions/missionsApi';

// import categoriesReducer from './categories/categories_duck';

const reducer = {
  rockets: rocketsSlice,
  missions: missionsSlice,
};

const store = configureStore({
  reducer,
  devTools: process.env.NODE_ENV !== 'production',
});

export default store;
