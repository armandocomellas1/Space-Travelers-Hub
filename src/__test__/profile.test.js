import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../Redux/store';
import MyProfile from '../components/MyProfile';

describe('test if MyProfile page works correctly', () => {
  test('renders correctly', () => {
    const profile = renderer.create(
      <Provider store={store}>
        <MyProfile />
      </Provider>,
    );
    expect(profile).toMatchSnapshot();
  });
});
