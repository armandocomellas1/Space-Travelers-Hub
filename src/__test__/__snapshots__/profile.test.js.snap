// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test if MyProfile page works correctly renders correctly 1`] = `null`;
